frame/merkle-mountain-range/primitives/src/lib.rs

```diff
@@ -26,6 +26,9 @@ use sp_std::fmt;
 #[cfg(not(feature = "std"))]
 use sp_std::prelude::Vec;
 
+/// Node index type.
+pub type NodeIndex = u64;
```

Comment:
I feel this might be helping understanding the code, i.e. we expect not an arbitrary `u64`, but rather something that describes the `NodeIndex` - however I agree, this is questionable, hence it wasn't there in the initial implementation.

Reply:
Just like why you guys use `SessionIndex` instead of `u32`.

I think this is okay.

Also, there is another reason.
In most chains, the `BlockNumber` is `u32`.
And the leaf index is "BlockNumber" (in MMR data structure).
People will easily make a mistake here.